src/main/java/com/monumental/services/ModelService.java

```diff
@@ -190,4 +190,59 @@ public void delete(Integer id) throws HibernateException {
     public void delete(List<Integer> ids) throws HibernateException {
         this.doDelete(ids);
     }
+
```

Comment:
I added tests, and also in order to get them to work had to add a new layer of complexity to all of our functions in `ModelService` that do gets. They now all have an optional `boolean initializeLazyLoadedCollections` parameter that when true will cause them to try to initialize any lazy loaded collections on the records before the session is closed.

Once the session is closed you are no longer able to initialize these so it has to be done here. You also can't just open a new session and do it. We also must close the sessions here or else we will hit connection limits on the database since none of our sessions will be getting closed.

Reply:
I also had to flip the placement of the `@ManyToMany` in `Monument` and `Tag` so that when you can insert a `Tag` with `Monument` references. For some reason you can only do this on the model that has the `@JoinTable` annotation on it and I doubt we will be inserting `Monument`s with `Tag`s very often but we will always be inserting `Tag`s with `Monument`s